app/helpers/CacheMiddleware.php

```diff
@@ -64,15 +69,22 @@ private function pathToCacheKey($path)
         return $path;
     }
 
-    private function shouldCache($req)
-    {
+    private function shouldCache(Request $req) {
         return $req->isGet();
     }
 
-    private function getCacheKey($req)
-    {
+    private function isJsonData(Request $req) {
+        return strpos($req->getUri()->getPath(), 'data/') !== false;
```

Comment:
I haven't tested it, but what if the URL contains `data/` as in `https://developer.piwik.org/guides/log-data/xxx`? This URL wouldn't work but there might be something in the future. I haven't looked yet at the structure of `data` URLs but maybe we can check if it starts with a certain URL or something else? Just a thought...

Reply:
You are right, that could cause issues in the future. I’ll replace it with something more specific

Reply:
This should be fixed now in 667cc0b04960e9f1387eb24d9a5216738928270d